legal-api/src/legal_api/resources/v2/namerequest.py

```diff
@@ -35,7 +38,23 @@ def get(identifier):
         if nr_response.status_code == 404:
             return make_response(jsonify(message='{} not found.'.format(identifier)), 404)
 
-        return jsonify(nr_response.json())
+        nr_json = nr_response.json()
+
+        # Check the NR is affiliated with this account
+        affiliate = AccountService.get_org_details_by_affiliation(identifier)
+        if affiliate.status_code == HTTPStatus.OK:
+            return jsonify(nr_json)
+
+        # If NR is not affiliated, validate the email and phone
+        email = request.args.get('email', None)
+        phone = request.args.get('phone', None)
+        nr_phone = nr_json.get('applicants').get('phoneNumber')
+        nr_email = nr_json.get('applicants').get('emailAddress')
+
+        if (phone and phone != nr_phone) or (email and email != nr_email):
+            return make_response(jsonify(message='Invalid email or phone number.'), 400)
```

Comment:
HTTP 400 is "bad request". What does the other NR endpoint (Namex or Auth API) return if unauthorized -- 401 ? 403 ?

Reply:
I agree with HTTP 400 on line 69.

I think this return code should be HTTP 403. @argush3 , thoughts?

Reply:
403 makes sense. updated